pkg/morph/event/listener.go

```diff
@@ -175,70 +187,47 @@ func (l *listener) listen(ctx context.Context, intError chan<- error) error {
 
 		hashes = append(hashes, hashType.ScriptHash())
 	}
-
-	// mark listener as started
-	l.started = true
-
 	l.mtx.RUnlock()
 
-	chEvent, err := l.subscriber.SubscribeForNotification(hashes...)
+	err := l.subscriber.SubscribeForNotification(hashes...)
 	if err != nil {
-		return err
+		errCh <- fmt.Errorf("could not subscribe for notifications: %w", err)
+		return
 	}
 
-	l.listenLoop(ctx, chEvent, intError)
-
-	return nil
-}
-
-func (l *listener) listenLoop(ctx context.Context, chEvent <-chan *state.ContainedNotificationEvent, intErr chan<- error) {
-	var (
-		blockChan <-chan *block.Block
-
-		notaryChan <-chan *result.NotaryRequestEvent
-
-		err error
-	)
-
 	if len(l.blockHandlers) > 0 {
-		if blockChan, err = l.subscriber.BlockNotifications(); err != nil {
-			if intErr != nil {
-				intErr <- fmt.Errorf("could not open block notifications channel: %w", err)
-			} else {
-				l.log.Debug("could not open block notifications channel",
-					zap.String("error", err.Error()),
-				)
-			}
-
+		if err = l.subscriber.BlockNotifications(); err != nil {
+			errCh <- fmt.Errorf("could not subscribe for blocks: %w", err)
 			return
 		}
-	} else {
-		blockChan = make(chan *block.Block)
 	}
 
 	if l.listenNotary {
-		if notaryChan, err = l.subscriber.SubscribeForNotaryRequests(l.notaryMainTXSigner); err != nil {
-			if intErr != nil {
-				intErr <- fmt.Errorf("could not open notary notifications channel: %w", err)
-			} else {
-				l.log.Debug("could not open notary notifications channel",
-					zap.String("error", err.Error()),
-				)
-			}
-
+		if err = l.subscriber.SubscribeForNotaryRequests(l.notaryMainTXSigner); err != nil {
+			errCh <- fmt.Errorf("could not subscribe for notary requests: %w", err)
 			return
 		}
 	}
+}
+
+func (l *listener) listenLoop(ctx context.Context, intErr chan<- error, subErrCh chan error) {
+	chs := l.subscriber.NotificationChannels()
 
 loop:
 	for {
 		select {
+		case err := <-subErrCh:
+			if intErr != nil {
+				intErr <- err
+			}
```

Comment:
We had a log in the `else` branch in the previous implementation. Why do we not have it now?

Reply:
added `Error` (not `Debug` like was before) log